### PR TITLE
Type contracts

### PR DESCRIPTION
**Pull request by therealemjy**
- add support for TypeChain
- apply generated types on contract getters and hooks
- fix type errors in the codebase following this change

### CODE REVIEW

File: src/clients/contracts/getters.ts [hunk 1]
```diff
@@ -10,79 +12,161 @@ import vaiUnitrollerAbi from 'constants/contracts/abis/vaiUnitroller.json';
...
+    return getContract(xvsTokenAbi as AbiItem[], tokenAddress, web3) as unknown as TokenContract<T>;
   }
 
-  return getContract(abi, getToken(name).address, web3);
+  if (tokenId === 'vai') {
+    return getContract(vaiTokenAbi as AbiItem[], tokenAddress, web3) as unknown as TokenContract<T>;
+  }
+
+  if (tokenId === 'vrt') {
+    return getContract(vrtTokenAbi as AbiItem[], tokenAddress, web3) as unknown as TokenContract<T>;
```
Comment (coreyar) [inline src/clients/contracts/getters.ts]: I think we could improve this by making `getContract` generic

Comment (therealemjy) [reply src/clients/contracts/getters.ts]: It is generic: you can get any contract with it (which is what we do in all getters in this file)

Comment (coreyar) [reply src/clients/contracts/getters.ts]: It does not have a generic type. This is what I was thinking 
```
const getContract = <T>(abi: AbiItem | AbiItem[], address: string, web3Instance: Web3) => {
  const web3 = web3Instance ?? getWeb3NoAccount();
  return new web3.eth.Contract(abi, address) as unknown as T;
};

export const getTokenContract = <T extends TokenId>(tokenId: T, web3: Web3): TokenContract<T> => {
  const tokenAddress = getToken(tokenId).address;

  if (tokenId === 'xvs') {
    return getContract<TokenContract<T>>(xvsTokenAbi as AbiItem[], tokenAddress, web3);
  }

  if (tokenId === 'vai') {
    return getContract<TokenContract<T>>(vaiTokenAbi as AbiItem[], tokenAddress, web3);
  }

  if (tokenId === 'vrt') {
    return getContract<TokenContract<T>>(vrtTokenAbi as AbiItem[], tokenAddress, web3);
  }
  ```

Comment (therealemjy) [reply src/clients/contracts/getters.ts]: Oh I see, I didn't understand that. Works for me!

**Review by coreyar — APPROVED** [review]

**Pull request merged**